test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -731,17 +731,31 @@ data Action' h a where
     => Var h (WrapTable h IO k v b)
     -> R.UnionCredits
     -> Act' h R.UnionCredits
-  -- TODO: SupplyUnionCredits has no information about union debt, so the
-  -- credits we generate are arbitrary, and it would require precarious,
-  -- manual tuning to make sure the debt is ever paid off by an action
-  -- sequence. We can probably add a version of SupplyUnionCredits that
-  -- supplies a percentage (if not al) of the current or previously computed
-  -- debt. Note that the model has no accurate debt information, so we might
-  -- have to use variables.
+  -- | Alternative version of 'SupplyUnionCredits' that supplies a portion of
```

Comment:
Maybe it could even replace `SupplyUnionCredits`. Supplying absolute credits doesn't seem super meaningful without a frame of reference (how much debt there is). The distribution of portions seems easier to tune, as it automatically scales with union size.

Although absolute credits could sometimes allow for better shrinking if a specific number of supplied credits triggers an issue, but shrinking any of the union inputs would make `SupplyPortionOfDebt` result in a different number of credits. :shrug:

Reply:
I think it's useful to keep `SupplyUnionCredits` because it broadens the "generator space" beyond supplying 1x 1/2x 1/3x the debt, and so we can shrink `SupplyPortionDebt` to `SupplyUnionCredits`. There are cases like you describe where shrinking to absolute credits might not work so well, but I also don't think it's entirely useless in cases where we are shrinking an independent action